### PR TITLE
PR GPS Reset

### PR DESCRIPTION
**Pull request by MatejFranceskin**
Solution for issue #26 Add UBX-CFG-RST to force a Cold Start

### CODE REVIEW

File: src/gps_helper.h [hunk 1]
```diff
@@ -151,6 +158,15 @@ class GPSHelper
 	 */
 	virtual int receive(unsigned timeout) = 0;
 
+	/**
+	 * Reset GPS device
+	 * @param restart_type
+	 * @return -1 if not implemented
+	 * 	        0 on failure
```
Comment (bkueng) [inline src/gps_helper.h]: Can you return 0 on success, and <0 for failures?

File: src/gps_helper.h [hunk 2]
```diff
@@ -98,6 +98,13 @@ enum class GPSCallbackType {
 	setClock,
 };
 
+enum class GPSRestartType {
+	None,
+	Hot,
```
Comment (bkueng) [inline src/gps_helper.h]: Can you add some documentation what the exact meaning is for each type?

File: src/ubx.h [hunk 3]
```diff
@@ -202,6 +204,15 @@
 #define UBX_TX_CFG_NAV5_MASK		0x0005		/**< Only update dynamic model and fix mode */
 #define UBX_TX_CFG_NAV5_FIXMODE		2		/**< 1 2D only, 2 3D only, 3 Auto 2D/3D */
 
+/* TX CFG-RST message contents
+ * Note: not used with protocol version 27+ anymore
```
Comment (bkueng) [inline src/ubx.h]: Can you add a TODO in the code or use the new message if `_proto_ver_27_or_higher` is set?

File: src/gps_helper.h (hunk 2 above)
Comment (MatejFranceskin) [reply src/gps_helper.h]: Added descriptions from UBX documentation.

File: src/gps_helper.h (hunk 1 above)
Comment (MatejFranceskin) [reply src/gps_helper.h]: Some GPS drivers do not have reset function, so I changed it like this:
0 : success
-1: not implemented
<0: failure

File: src/ubx.h (hunk 3 above)
Comment (MatejFranceskin) [reply src/ubx.h]: Removed the comment - it was a copy/paste error and it was not vaild for CFG-RST

**Comment by MatejFranceskin** [comment]
Thank you for the review - I improved the code and rebased so that everything is in sync with upstream.

File: src/gps_helper.h [hunk 4]
```diff
@@ -98,6 +98,33 @@ enum class GPSCallbackType {
...
+	 * In hot start mode, the receiver was powered down only for a short time (4 hours or less),
+	 * so that its ephemeris is still valid. Since the receiver doesn't need to download ephemeris
+	 * again, this is the fastest startup method.
+	 */
+	Hot,
+
+	/**
+	 * In warm start mode, the receiver has approximate information for time, position, and coarse
+	 * satellite position data (Almanac). In this mode, after power-up, the receiver normally needs
+	 * to downloadephemeris before it can calculate position and velocity data.
```
Comment (bkueng) [inline src/gps_helper.h]: typo: `downloadephemeris`

**Pull request closed without merging**